### DISCUSSION TITLE
[Suggestions]

### DISCUSSION THREAD

**Discussion by Mister-Hope** [ntnyq/vuepress-plugin-social-share#67] [Ideas]
1. Could you please open discussion panel?
2. It would be great if you can import styling tools (not the same as linter) like prettier to make your code style constant. As it would be much easier for others to create pr for you.
3. Some changes about Api are listed here:

## Merge network and extraNetwork

There is no need to provide `networks` and `extraNetworks` at same time. There could be only one option network:

```ts
// built-in keywords
type SupportedNetwork = 'twitter' | 'facebook' | ...  ;


export interface SocialSharePluginOptions {
  networks?: (SupportedNetwork  | SocialShareNetwork)[]
  // others...
  /** @deprecated use `networks` directly*/
 extraNetWorks?: SocialShareNetwork[]
}
```

Also, props on components could be updated in the same way. Meanwhile it's easy to add a legacy support.

## Update on frontmatter with primary option, as well as marking other options deprecated

It's better to create a scope for `social-share` in frontmatter, so the new frontmatter options could be:

```ts
import { type PageFrontmatter } from "vuepress";

export interface SocialShareFrontmatter {
  enable?: boolean;
  global?: boolean;

  // share meta
  /**
   * Share link
   *
   * @default window.location.href
   */
  url?: string;

  /**
   * Share title
   *
   * @default frontmatter.title
   */
  title?: string;

  /**
   * Share description
   *
   * @default frontmatter.description
   */
  description?: string;

  // ...
}

export interface SocialSharePluginFrontmatter extends PageFrontmatter {
  "social-share"?: SocialShareFrontmatter;

  /** @deprecated use `social-share.enable`  instead  */
  noSocialShare?: boolean;
  /** @deprecated use `social-share.global` instead  */
  noGlobalSocialShare?: boolean;

  // share meta
  /** @deprecated use `social-share.url` instead */
  shareUrl?: string;
  /** @deprecated use `social-share.url` instead */
  $shareUrl?: string;

  /** @deprecated use `social-share.title` instead */
  shareTitle?: string;
  /** @deprecated use `social-share.title` instead */
  $shareTitle?: string;

  // others, should be similar
}
```

## Better style customization and provide `custom` option

I do think that the default style does not look well, it would be greate to provide some style customization by allowing to override share-button svg through css variables, as well as a new option `custom` which allows themes or users to provide styles themselves.

**Comment by ntnyq** [comment]
1. The dIscussion panel is opened.
2. I'll add prettier to the project.

**Comment by Mister-Hope** [comment]
What about the other suggestions, also I would recommend you to convert this issue to discussion

**Reply by ntnyq** [reply]
Done.

**Comment by Mister-Hope** [comment]
I will open a few prs later, but I think I might still need to write something on my own.

**Comment by ntnyq** [comment]
Prettier is setup in the repo.

I have a full-time job in daytime and will think about the proposals above once there is some spared time.
